src/ai/susi/server/api/aaa/UploadAvatarService.java

```diff
@@ -0,0 +1,180 @@
+                    String imagePath = DAO.data_dir  + File.separator + "avatar_uploads";
+                    // Reading content for image
+                    Image image = ImageIO.read(imagePartContent);
+                    BufferedImage bi = UploadAvatarService.toBufferedImage(image);
+                    // Checks if images directory exists or not. If not then create one
+                    if (!Files.exists(Paths.get(imagePath))) new File(imagePath).mkdirs();
+
+                    String image_name = getMD5Hash(userEmail);
+                    File p = new File(imagePath + File.separator + image_name);
+                    if (p.exists()) p.delete();
+                    ImageIO.write(bi, "jpg", new File(imagePath + File.separator + image_name));
+                    result.put("accepted", true);
+                    result.put("image_path", image_name);
+                }
+                else{
+                    result.put("accepted", false);
+                    result.put("message", "Access token is invalid");
+                }
+            }
+            resp.setContentType("application/json");
+            resp.setCharacterEncoding("UTF-8");
+            resp.getWriter().write(result.toString());
+        }
+        else{
+            result.put("message","Access token are not given");
+            result.put("accepted",false);
+            resp.setContentType("application/json");
+            resp.setCharacterEncoding("UTF-8");
+            resp.getWriter().write(result.toString());
+        }
+    }
+    /**
+     * Converts a given Image into a BufferedImage
+     *
+     * @param img The Image to be converted
+     * @return The converted BufferedImage
+     */
+    public static BufferedImage toBufferedImage(Image img)
+    {
+        if (img instanceof BufferedImage)
+        {
+            return (BufferedImage) img;
+        }
+
+        // Create a buffered image with transparency
+        BufferedImage bimage = new BufferedImage(img.getWidth(null), img.getHeight(null), BufferedImage.TYPE_INT_ARGB);
+        // Draw the image on to the buffered image
+        Graphics2D bGr = bimage.createGraphics();
+        bGr.drawImage(img, 0, 0, null);
+        bGr.dispose();
+        // Return the buffered image
+        return bimage;
+    }
+    /**
+     * Utility method to get file name from HTTP header content-disposition
+     */
+    private static String getFileName(Part part) {
+        String contentDisp = part.getHeader("content-disposition");
+        System.out.println("content-disposition header= "+contentDisp);
+        String[] tokens = contentDisp.split(";");
+        for (String token : tokens) {
+            if (token.trim().startsWith("filename")) {
+                return token.substring(token.indexOf("=") + 2, token.length()-1);
+            }
+        }
+        return "";
+    }
+
+    /**
+     * Utility method to get MD5 hash of a string
+     */
+    private static String getMD5Hash (String inputText) {
```

Comment:
Don't implement the MD5 algorithm here because we have that already in Digest.encodeMD5Hex().

Reply:
furthermore: why do you use the MD5 of the userEmail? We already have an ID which is a hash that is passed in identity.getUuid(). That is in your current variable `userId`. That is the unique identifier for the user, thus that is the most appropriate name of the ìmage_name`.

Reply:
So I suggest that you omit the MD5 method and the encoding to MD5 completely, just ise the `userId`

Reply:
I had encoded the email to MD5 hash, because the same implementation has been done by Gravatar and both the implementation could be kept similar.

Reply:
@hkedia321 I think a separate endpoint for uploading avatar should be made. Also, it should be under the `/aaa` folder unlike the `/cms/uploadImage.json` endpoint.